### PR TITLE
debug: filter exceptions from DAP telemetry on web

### PR DESCRIPTION
**Pull request by connor4312**
This change will filter DAP keys beginning with `!` when the remote
authority is "other", which SteVen recommended as a way to tell if we're
on web/codespaces.

Isidor mentioned previously that there was filtering for keys beginning
with an underscore, but this was only applied to some data in error
responses, not general telemetry events. An alternative approach to this
is filtering exceptions based on the environment on the extension side,
but I figured it would be better to have VS Code do that so that we
don't end up with N many possibly deviant sets of environment detection
logic.

### CODE REVIEW

File: src/vs/workbench/contrib/debug/browser/debugSession.ts [hunk 1]
```diff
@@ -848,7 +848,17 @@ export class DebugSession implements IDebugSession {
 					// and the user opted in telemetry
 					if (this.raw.customTelemetryService && this.telemetryService.isOptedIn) {
 						// __GDPR__TODO__ We're sending events in the name of the debug extension and we can not ensure that those are declared correctly.
-						this.raw.customTelemetryService.publicLog(event.body.output, event.body.data);
+						let data = event.body.data;
```
Comment (isidorn) [inline src/vs/workbench/contrib/debug/browser/debugSession.ts:851]: Does it make sense to move this out to a helper method which would live in `debugTelemetry.ts`?

**Review by isidorn — APPROVED** [review]

**Comment by isidorn** [comment]
Looks good, great job 👏 
Just left one minor comment, if you can tackle it great, if not no problem.

File: src/vs/workbench/contrib/debug/browser/debugSession.ts (hunk 1 above)
Comment (connor4312) [reply src/vs/workbench/contrib/debug/browser/debugSession.ts:852]: The DebugTelemetry class isn't very accessible here; that's used for the top-level of the session and send to the global(?) telemetry service, and for this bit of telemetry we're sending to a separate instance of the service created from the `aiKey` inside the debug extension's package.json.

So perhaps not a method, but I did pop this to its own helper function in debugUtils.

Comment (isidorn) [reply src/vs/workbench/contrib/debug/browser/debugSession.ts:852]: Thanks. I like that!

**Pull request merged**